app/src/main/java/org/fossasia/openevent/app/core/auth/login/LoginFragment.java

```diff
@@ -49,25 +49,50 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     @Override
     public void onStart() {
         super.onStart();
-        getPresenter().attach(this);
-        binding.setLogin(getPresenter().getLogin());
-        getPresenter().start();
+
+        //attaching the view model
+        loginFragmentViewModel = ViewModelProviders.of(this, viewModelFactory).get(LoginViewModel.class);
+        handleNewIntent();
+        handleEmail();
+        binding.setLogin(loginFragmentViewModel.getLogin());
         binding.btnLogin.setOnClickListener(view -> {
             if (!validator.validate())
                 return;
 
             ViewUtils.hideKeyboard(view);
-
             String url = binding.url.baseUrl.getText().toString().trim();
-            getPresenter().setBaseUrl(url, binding.url.overrideUrl.isChecked());
-            getPresenter().login();
+
+            loginFragmentViewModel.setBaseUrl(url, binding.url.overrideUrl.isChecked());
+            loginFragmentViewModel.showProgress().observe(this, this::showProgress);
+            loginFragmentViewModel.login();
+            handleNewIntent();
+            loginFragmentViewModel.showError().observe(this, this::showError);
         });
 
         binding.signUpLink.setOnClickListener(view -> openSignUpPage());
-
         binding.forgotPasswordLink.setOnClickListener(view -> openForgotPasswordPage());
     }
 
+    private void handleEmail() {
+        loginFragmentViewModel.getEmailList().observe(this, emailList -> {
+            binding.emailDropdown.setAdapter(
+                new ArrayAdapter<>(getActivity(), android.R.layout.simple_list_item_1, new ArrayList<>(emailList))
+            );
+        });
+    }
+
+    public void handleNewIntent() {
+        loginFragmentViewModel.getLoginStatus().observe(this, isLoggedIn -> {
+                if(!isLoggedIn)
+                    return;
```

Comment:
No check should be placed here, simply push the data in livedata only if isLoggedIn is true